
###########################################################################################################################################################
                                                                   #LINUX PREPROCESSING                                                                                                                                
###########################################################################################################################################################


# Set the working directory
cd /PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data

# The merging of three reads per sample was performed using R 4.3.0 (https://github.com/celiatorresvilanova/TFM/blob/main/Preprocessing_R.r). Subsequently, 
# FastQC reports were generated for the merged samples

mkdir merged/Report_merged
fastqc -o merged/Reports_merged merged/*.fastq.gz

######################################################################### TRIMMING #########################################################################

# The samples were analyzed using Illumina NextSeq2000, with the following parameters: single-end reads, stranded, and 50 base pairs in length. To remove
# low-quality reads, sickle was performed, applying the following criteria: quality score (q 20), read length (l 50), single-end reads (se), and sanger
# quality encoding (t sanger). The adoption of sanger quality encoding by Illumina since 2012 requires the specification of -t sanger during the analysis.

for file in merged/*.fastq.gz
     # The output filename was generated by replacing the directory path with merged_quality and removing the .gz extension
     output_file="merged_quality/$(basename ${file} .fastq.gz).fastq"

     # The input file was decompressed and sickle was performed
     gunzip -c "$file" | sickle se -f /dev/stdin -t sanger -o "$output_file" -q 20 -l 50
done

# The final outputs were compressed
cd merged_quality
gzip *.fastq

# FastQC reports were generated for the merged samples after trimming
fastqc -o Reports_merged_quality merged_quality/*.fastq.gz

# A high ratio of duplicates was observed in both the merged files and the merged files after trimming. Duplicate management will be handled post-alignment.

####################################################################### STAR ALIGNMENT #######################################################################

# The files for v43 of the human genome fasta and annotation were downloaded from GENCODE, and then the files were decompressed
gunzip gencode.v43.primary_assembly.basic.annotation.gtf.gz
gunzip GRCh38.primary_assembly-genome.fa GRCh38.fa.gz

# The annotation and genome filenames were changed
mv gencode.v43.primary_assembly.basic.annotation.gtf GRCh38.gtf
mv GRCh38.primary_assembly-genome.fa GRCh38.fa

# Change the sample filenames to make them shorter
cd merged_quality

for file in *.fastq.gz; do
    new_name=$(echo "$file" | sed -E 's/^([^_]+_[^_]+)_.*(\.fastq\.gz)$/\1\2/')
    mv "$file" "$new_name"
done

# A genome index was build taking into accoung the genome and annotation files
mkdir Alignments

STAR --runThreadN 30 \
     --runMode genomeGenerate \
     --genomeDir merged_quality/Alignments/genomeIndex \
     --genomeFastaFiles GRCh38.fa \
     --sjdbGTFfile GRCh38.gtf

# The STAR alignment was conducted with the following parameters:
     # The ReadFilesIn argument separated filenames with commas, as single-end reads are recognized in this manner by the STAR algorithm 
     # The OutSAMunmapped Within argument directed unmapped reads to be written to a separate section in the output file
     # The OutSAMattributes argument specified the attributes to be included in the alignment output: number of hits (NH), alignment hit index (HI), alignment score (AS), number of mismatches (NM) and mismatching positions and lengths (MD)
     # The OutSAMattrRGline argument allowed for specifying sample information for the alignment
     # The ReadFilesCommand argument decompressed the input files
     # The OutSAMtype argument determined the output format and sorting order for the alignment output

STAR --runThreadN 30 \
     --genomeDir Alignments/genomeIndex \
     --readFilesIn S10_1.fastq.gz,S12_2.fastq.gz,S14_2.fastq.gz,S17_1.fastq.gz,S19_3.fastq.gz,S22_1.fastq.gz,S24_1.fastq.gz,S3_3.fastq.gz,S6_2.fastq.gz,S10_2.fastq.gz,S12_3.fastq.gz,S14_3.fastq.gz,S17_2.fastq.gz,S20_1.fastq.gz,S22_2.fastq.gz,S24_2.fastq.gz,S4_1.fastq.gz,S6_3.fastq.gz,S10_3.fastq.gz,S1_2.fastq.gz,S15_1.fastq.gz,S17_3.fastq.gz,S20_2.fastq.gz,S22_3.fastq.gz,S24_3.fastq.gz,S4_2.fastq.gz,S7_1.fastq.gz,S11_1.fastq.gz,S13_1.fastq.gz,S15_2.fastq.gz,S18_1.fastq.gz,S20_3.fastq.gz,S2_2.fastq.gz,S25_1.fastq.gz,S4_3.fastq.gz,S7_2.fastq.gz,S11_2.fastq.gz,S13_2.fastq.gz,S15_3.fastq.gz,S18_2.fastq.gz,S21_1.fastq.gz,S23_1.fastq.gz,S25_2.fastq.gz,S5_1.fastq.gz,S7_3.fastq.gz,S11_3.fastq.gz,S13_3.fastq.gz,S16_1.fastq.gz,S18_3.fastq.gz,S21_2.fastq.gz,S23_2.fastq.gz,S25_3.fastq.gz,S5_2.fastq.gz,S9_1.fastq.gz,S1_1.fastq.gz,S1_3.fastq.gz,S16_2.fastq.gz,S19_1.fastq.gz,S21_3.fastq.gz,S23_3.fastq.gz,S3_1.fastq.gz,S5_3.fastq.gz,S9_2.fastq.gz,S12_1.fastq.gz,S14_1.fastq.gz,S16_3.fastq.gz,S19_2.fastq.gz,S2_1.fastq.gz,S2_3.fastq.gz,S3_2.fastq.gz,S6_1.fastq.gz,S9_3.fastq.gz \
     --outSAMunmapped Within \
     --outSAMattributes NH HI AS NM MD \
     --outSAMattrRGline ID:S10_1 , ID:S12_2 , ID:S14_2 , ID:S17_1 , ID:S19_3 , ID:S22_1 , ID:S24_1 , ID:S3_3 , ID:S6_2 , ID:S10_2 , ID:S12_3 , ID:S14_3 , ID:S17_2 , ID:S20_1 , ID:S22_2 , ID:S24_2 , ID:S4_1 , ID:S6_3 , ID:S10_3 , ID:S1_2 , ID:S15_1 , ID:S17_3 , ID:S20_2 , ID:S22_3 , ID:S24_3 , ID:S4_2 , ID:S7_1 , ID:S11_1 , ID:S13_1 , ID:S15_2 , ID:S18_1 , ID:S20_3 , ID:S2_2 , ID:S25_1 , ID:S4_3 , ID:S7_2 , ID:S11_2 , ID:S13_2 , ID:S15_3 , ID:S18_2 , ID:S21_1 , ID:S23_1 , ID:S25_2 , ID:S5_1 , ID:S7_3 , ID:S11_3 , ID:S13_3 , ID:S16_1 , ID:S18_3 , ID:S21_2 , ID:S23_2 , ID:S25_3 , ID:S5_2 , ID:S9_1 , ID:S1_1 , ID:S1_3 , ID:S16_2 , ID:S19_1 , ID:S21_3 , ID:S23_3 , ID:S3_1 , ID:S5_3 , ID:S9_2 , ID:S12_1 , ID:S14_1 , ID:S16_3 , ID:S19_2 , ID:S2_1 , ID:S2_3 , ID:S3_2 , ID:S6_1 , ID:S9_3 \
     --readFilesCommand pigz -p32 -dc \
     --outSAMtype BAM SortedByCoordinate \
     --outFileNamePrefix Alignments/

# The output file were checked to ensure that the IDs and the specified attributes are included
samtools view -h Alignments/Aligned.sortedByCoord.out.bam | more

# The mapping statistics of STAR alignment were checked
cat Alignments/Aligned.sortedByCoord.out.Log.final.out


                                        #                                Started job on |       May 29 13:20:54
                                        #                            Started mapping on |       May 29 13:25:34
                                        #                                   Finished on |       May 29 21:12:02
                                        #      Mapping speed, Million of reads per hour |       433.58

                                        #                         Number of input reads |       3370842019
                                        #                     Average input read length |       50
                                        #                                   UNIQUE READS:
                                        #                  Uniquely mapped reads number |       2927016651
                                        #                       Uniquely mapped reads % |       86.83%
                                        #                         Average mapped length |       49.89
                                        #                      Number of splices: Total |       432051703
                                        #           Number of splices: Annotated (sjdb) |       425276145
                                        #                      Number of splices: GT/AG |       428172907
                                        #                      Number of splices: GC/AG |       2865854
                                        #                      Number of splices: AT/AC |       374986
                                        #              Number of splices: Non-canonical |       637956
                                        #                     Mismatch rate per base, % |       0.24%
                                        #                        Deletion rate per base |       0.01%
                                        #                       Deletion average length |       1.35
                                        #                       Insertion rate per base |       0.00%
                                        #                      Insertion average length |       1.25
                                        #                            MULTI-MAPPING READS:
                                        #       Number of reads mapped to multiple loci |       387143226
                                        #            % of reads mapped to multiple loci |       11.49%
                                        #       Number of reads mapped to too many loci |       16489618
                                        #            % of reads mapped to too many loci |       0.49%
                                        #                                 UNMAPPED READS:
                                        # Number of reads unmapped: too many mismatches |       0
                                        #      % of reads unmapped: too many mismatches |       0.00%
                                        #           Number of reads unmapped: too short |       25110939
                                        #                % of reads unmapped: too short |       0.74%
                                        #               Number of reads unmapped: other |       15081585
                                        #                    % of reads unmapped: other |       0.45%
                                        #                                 CHIMERIC READS:
                                        #                      Number of chimeric reads |       0
                                        #                           % of chimeric reads |       0.00%



######################################################################### DUPLICATIONS #########################################################################

# In RNA-Seq, distinguishing between artificial reads and normal read duplication caused by over-sequencing highly expressed genes is challenging. This is because
# duplicate reads are commonly observed in RNA-Seq, and it is not exclusively attributed to technical errors. Highly expressed genes often surpass the threshold of 
# one read per base pair of the exon model, resulting in unavoidable read duplication. To handle this situation, I will mark the duplicates using Sambamba and then 
# perform dupRadar in R 4.3.0 (https://github.com/celiatorresvilanova/TFM/blob/main/Preprocessing_R.r). DupRadar is a tool that allows visualizing the relationship
#  between the percentage of duplications and gene expression. 

sambamba markdup -t 32 -p --tmpdir Duplicates Alignments/Aligned.sortedByCoord.out.bam  Duplicates/Aligned.sortedByCoord.out.mark.bam

# By using dupRadar, elevated duplication percentages were obtained, even in lowly expressed genes. Therefore, the decision was made to perform the analysis with two
# datasets: the original data and the data with duplicates removed. The duplicates were removed using Sambamba, specifying the argument -r (remove duplicates).

sambamba markdup -t 32 -r -p --tmpdir Duplicates Alignments/Aligned.sortedByCoord.out.bam  Duplicates/Aligned.sortedByCoord.out.dup.bam

######################################################################### QUANTIFICATION #########################################################################

# It is necessary to check the sorting performed by Sambamba, as this information is needed for quantification purposes. In this case, it is sorted by position.
samtools view -H Duplicates/Aligned.sortedByCoord.out.mark.bam | grep "@HD"  

# Better quantification with an index of the BAM file is achieved by Htseq. It needs to be performed only on the output of STAR, as the index file has already been
# generated by Sambamba)

samtools index Alignments/Aligned.sortedByCoord.out.bam

# The htseq-count quantification was conducted with the following parameters:
     # The -r argument specifies the alignment mode. In this case, was set to "pos", because the reads were aligned in a strand-speciffic manner
     # The -s argument specifies wheter the library is strans-specific or not. In this case is strand-specific (yes)
     # The -i argument specifies the feature type used for counting. In this case, it was set to gene_id, because the counting was performed at the gene level
     # The -t atgument specifies the feature type attribute. In this case, it was set to gene indicating that the counting is performed at the gene level.

mkdir Quantifications

htseq-count -f bam -r pos -s yes -i gene_id -t gene -n 32 \
--additional-attr ID \
Alignments/Aligned.sortedByCoord.out.bam \
../GRCh38.gtf  > Quantifications/raw.counts.matrix.txt

htseq-count -f bam -r pos -s yes -i gene_id -t gene -n 32 \
--additional-attr ID \
Duplicates/Aligned.sortedByCoord.out.dup.bam \
../GRCh38.gtf  > Quantifications/raw.counts.matrix.dup.txt

# The outputs were compressed
gzip Quantifications/raw.counts.matrix.txt
gzip Quantifications/raw.counts.dup.matrix.txt






