---
title: "SAINT"
author: "CÃ¨lia Torres Vilanova"
date: "2023-05-09"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(dupRadar)
library(stringr)
library(gtools)
library(Rsubread)
library(dplyr)
library(Biobase)
library(ggplot2)
library(plotly)

```

# 1. DupRadar 
## 1.1 Comprehensive Duplication Analysis of all combined Samples

```{r}

bamDuprm <- "/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/merged_quality/Duplicates/Aligned.sortedByCoord.out.mark.bam"	# the duplicate marked bam file
gtf <- "/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/GRCh38.gtf"	# the gene model
stranded <- 1		# '0' (unstranded), '1' (stranded) and '2' (reversely stranded)
paired   <- FALSE	# is the library paired end?
threads  <- 32		# number of threads to be used

dm.all.combined.samples <- analyzeDuprates(bamDuprm,gtf,stranded,paired,threads)
save(dm.all.combined.samples, file = "dm.all.RData")

par(mfrow=c(1,2))
duprateExpDensPlot(DupMat=dm.all.combined.samples)

# Figure S1.1 Duplication analysis on original data 

```

## 1.1.1 Filtering criteria to remove lowly expressed genes that have high duplicate rates

```{r}
dim(dm.all.combined.samples)

dm_f1 <- dm.all.combined.samples [!(dm.all.combined.samples$dupRateMulti>0.5 & dm.all.combined.samples$RPKMulti<10),]
dim(dm_f1)

dm_f2 <- dm_f1[!(dm_f1$dupRateMulti>0.7 & dm_f1$RPKMulti>=10 & dm_f1$RPKMulti<=100),]
dim(dm_f2)

dm_f3 <- dm_f2[!(dm_f2$dupRateMulti>0.75 & dm_f2$RPKMulti>=100 & dm_f2$RPKMulti<=500),]
dim(dm_f3)

dm_final <- dm_f3[!(dm_f3$dupRateMulti>0.8 & dm_f3$RPKMulti>=500 & dm_f3$RPKMulti<=1000),]
dim(dm_final)

save(dm_final, file = "dm_final.RData")

```



## 1.2. Separate Duplication Analysis of all samples

```{r}
bamDir <- "/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/merged_quality/Duplicates/Samples/"
gtf <- "/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/GRCh38.gtf"
stranded <- 1
paired <- FALSE
threads <- 32

bamFiles <- list.files(bamDir, pattern = "\\.mark\\.bam$", full.names = TRUE)

# Set the desired figure size
pdf("dupRadar_plots.pdf", width = 12, height = 8)

for (bamFile in bamFiles) {
  dm.separated.samples <- analyzeDuprates(bamFile, gtf, stranded, paired, threads)
  sampleName <- tools::file_path_sans_ext(basename(bamFile))
  par(mfrow = c(1, 2))
  duprateExpDensPlot(DupMat = dm.separated.samples)
  title(main = sampleName)
}

# Close the PDF device
dev.off()
```


## 1.3 Separate Duplication Analysis of problematic samples
### 1.3.1 First run

```{r}
bamDir <- "/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/merged_quality/Duplicates/Problematic_samples/1st_read"
gtf <- "/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/GRCh38.gtf"
stranded <- 1
paired <- FALSE
threads <- 32

bamFiles <- list.files(bamDir, full.names = TRUE)

# Set the desired figure size
pdf("problematic_samples_1st_read_dupRadar_plots.pdf", width = 12, height = 8)

for (bamFile in bamFiles) {
  dm.separated.samples <- analyzeDuprates(bamFile, gtf, stranded, paired, threads)
  sampleName <- tools::file_path_sans_ext(basename(bamFile))
  par(mfrow = c(1, 2))
  duprateExpDensPlot(DupMat = dm.separated.samples)
  title(main = sampleName)
}

# Close the PDF device
dev.off()
```

### 1.3.2 Second run

```{r}
bamDir <- "/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/merged_quality/Duplicates/Problematic_samples/2nd_read"
gtf <- "/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/GRCh38.gtf"
stranded <- 1
paired <- FALSE
threads <- 32

bamFiles <- list.files(bamDir, full.names = TRUE)

# Set the desired figure size
pdf("problematic_samples_2nd_read_dupRadar_plots.pdf", width = 12, height = 8)

for (bamFile in bamFiles) {
  dm.separated.samples <- analyzeDuprates(bamFile, gtf, stranded, paired, threads)
  sampleName <- tools::file_path_sans_ext(basename(bamFile))
  par(mfrow = c(1, 2))
  duprateExpDensPlot(DupMat = dm.separated.samples)
  title(main = sampleName)
}

# Close the PDF device
dev.off()
```

### 1.3.3 Third run

```{r}
bamDir <- "/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/merged_quality/Duplicates/Problematic_samples/3rd_read"
gtf <- "/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/GRCh38.gtf"
stranded <- 1
paired <- FALSE
threads <- 32

bamFiles <- list.files(bamDir, full.names = TRUE)

# Set the desired figure size
pdf("problematic_samples_3rd_read_dupRadar_plots.pdf", width = 12, height = 8)

for (bamFile in bamFiles) {
  dm.separated.samples <- analyzeDuprates(bamFile, gtf, stranded, paired, threads)
  sampleName <- tools::file_path_sans_ext(basename(bamFile))
  par(mfrow = c(1, 2))
  duprateExpDensPlot(DupMat = dm.separated.samples)
  title(main = sampleName)
}

# Close the PDF device
dev.off()
```

# 2. Merge counts files into a single counts matrix

High levels of duplication were observed in the data based on the FastQC reports and the graphs illustrating the duplicate ratio versus gene expression. Due to our limited understanding of the impact of duplicates on the data, quality control and preprocessing steps will be carried out for both the original dataset and the dataset excluding duplicates.

## 2.1 Original data

```{r}
setwd("/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/merged_quality/Quantifications")

# Get the list of counts files
counts_files_orig <- list.files(path = "/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/merged_quality/Quantifications", pattern = ".out.bam.txt")
file_names <- str_extract(counts_files_orig, "orig(.*)Aligned")
extract_names <- str_replace(file_names, "orig", "")
extract_names <- str_replace(extract_names, "Aligned", "")

# Read the first counts file to initialize the merged counts matrix
merged_counts_orig <- read.table(counts_files_orig[1], col.names = c("gene_id", extract_names[1]))

# Loop through the remaining counts files and merge with the existing counts matrix
for (i in 2:length(counts_files_orig)) {
  file <- read.table(counts_files_orig[i], col.names = c("gene_id", extract_names[i]))
  merged_counts_orig <- merge(merged_counts_orig, file, by = "gene_id", all = TRUE)
}

merged_counts_orig <- merged_counts_orig[-c(1:5),]
column_order <- c("gene_id", mixedsort(extract_names))
merged_counts_orig <- merged_counts_orig[,column_order]
dim(merged_counts_orig)

# Write the merged counts matrix to a file
write.table(merged_counts_orig, file = "Complete_matrix/complete_counts_matrix_orig.txt", sep = "\t", row.names = FALSE)

```


## 2.2 Data excluding all the duplicates

```{r}
setwd("/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/merged_quality/Quantifications")

# Get the list of counts files
counts_files <- list.files(path = "/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/merged_quality/Quantifications", pattern = ".dup.bam.txt")
file_names <- str_extract(counts_files, "dup(.*)Aligned")
extract_names <- str_replace(file_names, "dup", "")
extract_names <- str_replace(extract_names, "Aligned", "")

# Read the first counts file to initialize the merged counts matrix
merged_counts_dup <- read.table(counts_files[1], col.names = c("gene_id", extract_names[1]))

# Loop through the remaining counts files and merge with the existing counts matrix
for (i in 2:length(counts_files)) {
  file <- read.table(counts_files[i], col.names = c("gene_id", extract_names[i]))
  merged_counts_dup <- merge(merged_counts_dup, file, by = "gene_id", all = TRUE)
}

merged_counts_dup <- merged_counts_dup[-c(1:5),]
column_order <- c("gene_id", mixedsort(extract_names))
merged_counts_dup <- merged_counts_dup[,column_order]
dim(merged_counts_dup)

# Write the merged counts matrix to a file
write.table(merged_counts_dup, file = "Complete_matrix/complete_counts_matrix_dup.txt", sep = "\t", row.names = FALSE)

```

# 3. Load data and create an Expression Set
## 3.1 Original data 

```{r}
assay_data_orig <- read.delim("/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/merged_quality/Quantifications/Complete_matrix/complete_counts_matrix_orig.txt")
rownames(assay_data_orig) <- assay_data_orig$gene_id
assay_data_orig <- assay_data_orig[, -1]
save(assay_data_orig, file = "assay_data_orig.RData")

phenodata <- read.csv("/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/saint.csv")
phenodata<- phenodata[phenodata$day %in% c(1,4,7),]
phenodata$extraction.batch <- rep(c(1, 2, 3, 4, 5, 6), times = c(12, 12, 12, 12, 12, 12))
phenodata$available.ng <- c(684.79, 483.20, 690.37, 1000, 690.52, 777.42,
                            1000, 480.00, 1000, 1000, 1000, 910.00, 
                            1000, 1000, 1000, 1000, 1000, 1000,
                            1000, 1000, 1000, 1000, 343.85, 1000,
                            328.64, 931.68, 877.92, 1000, 1000, 615.22,
                            669.12, 1000, 1000, 1000, 1000, 598.58,
                            1000, 589.22, 833.35, 1000, 988.77, 1000,
                            522.58, 369.58, 447.78, 1000, 942.00, 891.33, 
                            294.15, 339.48, 1000, 1000, 1000, 1000, 
                            1000, 784.72, 647.02, 1000, 919.02, 999.26,
                            251.26, 1000, 1000, 1000, 1000, 1000,
                            1000, 1000, 703.44, 1000, 1000, 1000)

phenodata$quality.rna <- c(9.2, 9.3, 9.1, 8.5, 9.1, 9.4, 9.1, 9.7, 9.2, 9.1,
                           8.8, 9.1, 8.3, 8.9, 9.3, 8.8, 9.4, 9.2, 8.2, 8.8,
                           8.8, 8.9, 9, 9.2, 9, 9.2, 9.1, 8.7, 9.5, 9.2, 8.6,
                           8.7, 8.5, 7.5, 8.4, 9.1, 8.7, 9.2, 9.2, 9, 9, 9.2,
                           9.1, 9.5, 9.4, 8, 7.9, 8.8, 8.4, 8.4, 8.9, 9, 9.4,
                           9.2, 8.6, 9.2, 9.3, 8.7, 9, 8.8, 8.9, 9.2, 8.9, 8.8,
                           8.7, 7.9, 8.7, 8.7, 9.1, 8.7, 8.9, 9)

phenodata$nano260230 <- c(0.18, 0.12, 0.17, 0.27, 0.17, 0.09, NA, NA, NA, NA, 
                          NA, NA, 0.22, 0.14, 0.33, 0.21, 0.33, 0.27, 0.32, 
                          0.33, 0.30, 0.07, 0.03, 0.17, 0.09, 0.07, 0.19, 0.43,
                          0.25, 0.08, 0.06, 0.18, 0.15, 0.47, 0.14, 0.06, 0.13,
                          0.06, 0.08, 0.14, 0.06, 0.07, 0.10, 0.06, 0.07, 0.19,
                          0.08, 0.13, 0.06, 0.05, 0.30, 0.29, 0.10, 0.16, 0.13,
                          0.07, 0.06, 0.24, 0.16, 0.12, 0.02, 0.26, 0.28, 0.39,
                          0.26, 0.34, 0.16, 0.27, 0.04, 0.14, 0.15, 0.40)

phenodata$nano260280 <- c(2.46, 2.41, 2.31, 2.28, 2.65, 2.26, NA, NA, NA, NA, 
                          NA, NA, 2.18, 2.36, 2.14, 2.14, 2.22, 2.17, 2.23,
                          2.21, 2.15, 2.36, 3.22, 2.26, 2.80, 2.54, 2.36, 2.18,
                          2.36, 2.53, 2.56, 2.19, 2.18, 2.17, 2.29, 2.40, 2.33,
                          2.64, 2.47, 2.26, 2.54, 2.43, 2.85, 3.17, 2.88, 2.31,
                          2.76, 2.36, 3.67, 3.35, 2.21, 2.15, 2.40, 2.24, 2.28,
                          2.35, 2.45, 2.21, 2.42, 2.33, 4.90, 2.30, 2.17, 2.16,
                          2.17, 2.23, 2.35, 2.21, 2.68, 2.33, 2.16, 2.13)

phenodata$library.batch <- rep(c(1, 2), times = c(24,48))

save(phenodata, file="phenodata.RData")

rownames(phenodata) <- colnames(assay_data_orig)

ExpressionSet_orig <- ExpressionSet(assayData = as.matrix(assay_data_orig), phenoData = AnnotatedDataFrame(phenodata))

save(ExpressionSet_orig, file="ExpressionSet_orig.RData")

```


## 3.2 Data excluding lowly expressed genes that have high duplication ratios 

```{r}
intersection <- intersect(dm_final$ID, rownames(assay_data_orig))
length(intersection)

assay_data_filt_genes <- assay_data_orig[intersection,]
dim(assay_data_filt_genes)

ExpressionSet_filt_genes <- ExpressionSet(assayData = as.matrix(assay_data_filt_genes), phenoData = AnnotatedDataFrame(phenodata))

save(assay_data_filt_genes, file="assay_data_filt_genes.RData")
save(ExpressionSet_filt_genes, file="ExpressionSet_filt_genes.RData")

```

## 3.3 Data excluding duplicates 

```{r}

assay_data_dup <- read.delim("/PROJECTES/MALARIA_IMMUNO/22.SAINT/Transcriptomics/Data/merged_quality/Quantifications/Complete_matrix/complete_counts_matrix_dup.txt")
rownames(assay_data_dup) <- assay_data_dup$gene_id
assay_data_dup <- assay_data_dup[, -1]

ExpressionSet_dup <- ExpressionSet(assayData = as.matrix(assay_data_dup), phenoData = AnnotatedDataFrame(phenodata))

```


# 4. Filtration
## 4.1. Filtration by low expressed genes
### 4.1.1 Original data

```{r}
dim(assay_data_orig) # 62,757 genes
keep <- rowSums(assay_data_orig > 10) >= 12 # at least 12 samples have 10 reads per gene
assay_data_orig_f <- assay_data_orig[keep,]
dim(assay_data_orig_f)  # 20,729 genes
```

### 4.1.2 Data excluding lowly expressed genes that have high duplication ratios  

```{r}
dim(assay_data_filt_genes) #  47,173 genes
keep <- rowSums(assay_data_filt_genes > 10) >= 12 # at least 12 samples have 10 reads per gene
assay_data_filt_genes_f <- assay_data_filt_genes[keep,]
dim(assay_data_filt_genes_f)  # 19,980 genes
```

### 4.1.3 Data excluding duplicates

```{r}
dim(assay_data_dup) # 62,757 genes
keep <- rowSums(assay_data_dup > 10) >= 12 # at least 12 samples have 10 reads per gene
assay_data_dup_f <- assay_data_dup[keep,]
dim(assay_data_dup_f)  # 17,295 genes
```


## 4.2 Filtration of transcripts that translate to globin domains and chains
### 4.2.1 Original data

```{r} 
rownames(assay_data_orig_f) <- gsub ("\\..*", "", rownames(assay_data_orig_f))

globin_transcripts <- c("ENSG00000086506", "ENSG00000161544","ENSG00000165553", "ENSG00000188536", "ENSG00000206172", "ENSG00000206177", "ENSG00000213931", "ENSG00000213934", "ENSG00000223609", "ENSG00000244734", "ENSG00000260629","ENSG00000196565")

assay_data_orig_f <- assay_data_orig_f[!(rownames(assay_data_orig_f) %in% globin_transcripts), ]
dim(assay_data_orig_f) # 20,722 genes

save(assay_data_orig_f, file="assay_data_orig_f.RData")

```

### 4.2.2 Data excluding lowly expressed genes that have high duplication ratios  

```{r} 
rownames(assay_data_filt_genes_f) <- gsub ("\\..*", "", rownames(assay_data_filt_genes_f))

assay_data_filt_genes_f <- assay_data_filt_genes_f[!(rownames(assay_data_filt_genes_f) %in% globin_transcripts), ]
dim(assay_data_filt_genes_f) # 19,973 genes

save(assay_data_filt_genes_f, file="assay_data_filt_genes_f.RData")

```

### 4.2.3 Data excluding duplicates

```{r} 
rownames(assay_data_dup_f) <- gsub ("\\..*", "", rownames(assay_data_dup_f))

assay_data_dup_f <- assay_data_dup_f[!(rownames(assay_data_dup_f) %in% globin_transcripts), ]
dim(assay_data_dup_f) # 17,288 genes

```


# 5. Data Visualization
## 5.1 Total reads per sample 
### 5.1.1 Histogram of the distribution of total reads per sample
#### 5.1.1.1 Original data

```{r}

orig_palette <- c("#2C7DA0", "#013A63", "#2A6F97", rep("#2C7DA0", 2), "#2A6F97",
                  "#61A5C2", "#2C7DA0", rep("#2A6F97", 4), "#2C7DA0", 
                  rep("#2A6F97", 2), "#89C2D9", rep("#2A6F97", 2), "#89C2D9", 
                  rep("#2C7DA0", 3), rep("#2A6F97", 4), "#2C7DA0",
                  "#2A6F97", "#013A63", "#2C7DA0", "#61A5C2", "#2C7DA0",
                  "#2A6F97", "#2A6F97", rep("#2C7DA0", 2), "#014F86", "#61A5C2",
                  rep("#2A6F97", 4), "#014F86", "#2A6F97", "#014F86", "#2C7DA0",
                  rep("#2A6F97", 2), rep("#014F86", 2), "#013A63",
                  rep("#2C7DA0", 3), "#2A6F97", rep("#014F86", 2), "#61A5C2",
                  "#2C7DA0", "#014F86", "#2C7DA0", rep("#2A6F97", 2), "#2C7DA0",
                  "#2A6F97", "#014F86", "#2C7DA0", "#014F86", "#013A63", 
                  "#2A6F97", "#012A4A","#014F86")
 
sampleT_orig <- apply(assay_data_orig,2,sum)/10^6
sampleT_orig

orig_df <- data.frame(Sample = names(sampleT_orig), Value = sampleT_orig)

theme_set(theme_minimal(base_size = 15))

hist_orig <- ggplot(orig_df, aes(x = Value)) +
  geom_histogram(aes(fill = Sample), binwidth = 1, color = "black") +
  scale_fill_manual(values = orig_palette) +
  labs(title = "Figure S2.1.1 Histogram of reads per sample in the original data",
       x = "Reads per sample", y = "Frequency")

print(hist_orig)

```
#### 5.1.1.2 Data excluding lowly expressed genes that have high duplication ratios

```{r}
filt_genes_palette <- c("#2C7DA0", "#012A4A", "#2A6F97", rep("#2C7DA0", 2), "#2A6F97",
                  "#61A5C2", "#2C7DA0", rep("#2A6F97", 4), "#2C7DA0", 
                  rep("#2A6F97", 2), "#89C2D9", rep("#2A6F97", 2), "#89C2D9", 
                  rep("#2C7DA0", 3), rep("#2A6F97", 4), "#2C7DA0",
                  "#014F86", "#013A63", "#2C7DA0", "#61A5C2", "#2C7DA0",
                  "#2A6F97", "#2A6F97", rep("#2C7DA0", 2), "#014F86", "#2C7DA0",
                  rep("#2A6F97", 4), "#014F86", "#2A6F97", "#014F86", "#2C7DA0",
                  rep("#2A6F97", 2), rep("#014F86", 2), "#013A63",
                  rep("#2C7DA0", 3), "#2A6F97", rep("#014F86", 2), "#61A5C2",
                  "#2C7DA0", "#014F86", "#2C7DA0", "#014F86", "#2A6F97", "#2C7DA0",
                  "#2A6F97", "#014F86", "#2C7DA0", "#014F86", "#013A63", 
                  "#2A6F97", "#012A4A","#014F86")
 

sampleT_filt_genes <- apply(assay_data_filt_genes,2,sum)/10^6
sampleT_filt_genes

filt_genes_df <- data.frame(Sample = names(sampleT_filt_genes), Value = sampleT_filt_genes)

theme_set(theme_minimal(base_size = 15))

hist_filt_genes <- ggplot(filt_genes_df, aes(x = Value)) +
  geom_histogram(aes(fill = Sample), binwidth = 1, color = "black") +
  scale_fill_manual(values = filt_genes_palette) +
  labs(title = "Figure S2.1.2 Histogram of reads per sample in the data excluding lowly expressed genes that have high ratio of duplicates",
       x = "Reads per sample", y = "Frequency")

print(hist_filt_genes)
```

#### 5.1.1.3 Data excluding the duplicates 

```{r}

sampleT_dup <- apply(assay_data_dup, 2, sum) / 10^6
sampleT_dup

dup_palette <- c("#2C7DA0", "#014F86", "#2C7DA0", "#2A6F97", "#61A5C2",
                 "#89C2D9", "#61A5C2", "#89C2D9", "#61A5C2", "#2C7DA0",
                 "#468FAF", rep("#2C7DA0", 2), "#61A5C2", "#2C7DA0",
                 "#89C2D9", "#468FAF", "#61A5C2", "#89C2D9", 
                 rep("#A9D6E5",2), "#2C7DA0", "#01497C", "#2C7DA0",  
                 "#468FAF", "#2C7DA0", "#61A5C2", "#013A63", "#01497C", 
                 rep("#61A5C2", 2),  rep("#89C2D9", 2), rep("#468FAF", 2),
                 rep("#61A5C2", 2), "#468FAF", "#2C7DA0", "#468FAF", "#89C2D9",
                 "#61A5C2", "#013A63", "#89C2D9", "#A9D6E5", "#89C2D9",
                 "#2C7DA0", "#468FAF", "#2C7DA0", rep("#2A6F97", 2),
                 "#89C2D9", "#468FAF", rep("#61A5C2", 2), "#468FAF", "#61A5C2",
                 rep("#89C2D9", 3), "#468FAF", rep("#61A5C2", 4), "#2C7DA0",
                 rep("#61A5C2", 2), "#01497C", "#468FAF", "#012A4A", "#89C2D9")

dup_df <- data.frame(Sample = names(sampleT_dup), Value = sampleT_dup)

theme_set(theme_minimal(base_size = 15))

hist_dup <- ggplot(dup_df, aes(x = Value)) +
  geom_histogram(aes(fill = Sample), binwidth = 1, color = "black") +
  scale_fill_manual(values = dup_palette) +
  labs(title = "Figure S2.1.3 Histogram of reads per sample in the data excluding the duplicates",
       x = "Reads per sample", y = "Frequency")

print(hist_dup)


```

### 5.1.2 Boxplots of the total reads per sample
#### 5.1.2.1 Original data

```{r}

dataframe.orig <- data.frame(Value = sampleT_orig)

# Create the boxplot
boxplot.orig <- ggplot(dataframe.orig, aes(y = Value)) +
  geom_boxplot(fill = "#4590caff", color = "black") +
  labs(title = "Figure S2.2.1 Boxplot of the reads per sample in the original data",
       x = "Reads per sample", y = "Frequency")+
  theme(plot.title = element_text(size = 15, face = "bold"),
        axis.title = element_text(size = 12),
        axis.text = element_text(size = 10))

# Add labels for outliers
outliers.orig <- boxplot.stats(sampleT_orig)$out
outliers_orig_df <- data.frame(Value = outliers.orig, Label = names(outliers.orig))
boxplot.orig <- boxplot.orig + geom_text(data = outliers_orig_df, aes(x = 1, 
                                         y = Value, label = Label),
                                         color = "black", size = 7, 
                                         vjust = -0.5, nudge_y = -0.1,
                                         nudge_x = -0.96)

print(boxplot.orig)

```

#### 5.1.2.2 Data excluding lowly expressed genes that have high duplication ratios

```{r}

# Create a data frame for plotting
dataframe.filt.genes <- data.frame(Value = sampleT_filt_genes)

# Create the boxplot
boxplot.filt.genes <- ggplot(dataframe.filt.genes, aes(y = Value)) +
  geom_boxplot(fill = "#4590caff", color = "black") +
  labs(title = "Figure S2.2.2 Boxplot of the reads per sample in the data excluding lowly expressed genes that have high duplication ratios",
       x = "Reads per sample", y = "Frequency")+
  theme(plot.title = element_text(size = 15, face = "bold"),
        axis.title = element_text(size = 12),
        axis.text = element_text(size = 10))


# Add labels for outliers
outliers.filt.genes <- boxplot.stats(sampleT_filt_genes)$out
outliers_filt_genes_df <- data.frame(Value = outliers.filt.genes, Label = names(outliers.filt.genes))
boxplot.filt.genes <- boxplot.filt.genes + geom_text(data = outliers_filt_genes_df, aes(x = 1, 
                                         y = Value, label = Label),
                                         color = "black", size = 7, 
                                         vjust = -0.5, nudge_y = -0.05,
                                         nudge_x = -0.96)

print(boxplot.filt.genes)

```

#### 5.1.2.3 Data excluding the duplicates


```{r}

# Create a data frame for plotting
dataframe.dup <- data.frame(Value = sampleT_dup)

# Create the boxplot
boxplot.dup <- ggplot(dataframe.dup, aes(y = Value)) +
  geom_boxplot(fill = "#4590caff", color = "black") +
  labs(title = "Figure S2.2.3 Boxplot of the reads per sample in the data without duplicates",
       x = "Reads per sample", y = "Frequency")+
  theme(plot.title = element_text(size = 15, face = "bold"),
        axis.title = element_text(size = 12),
        axis.text = element_text(size = 10))


# Add labels for outliers
outliers.dup <- boxplot.stats(sampleT_dup)$out
outliers_dup_df <- data.frame(Value = outliers.dup, Label = names(outliers.dup))
boxplot.dup <- boxplot.dup + geom_text(data = outliers_dup_df, aes(x = 1, 
                                         y = Value, label = Label),
                                         color = "black", size = 7, 
                                         vjust = -0.5, nudge_y = -0.05,
                                         nudge_x = -0.96)

print(boxplot.dup)

```

## 5.2 Duplicates 
### 5.2.1 Principal Component Analysis of the duplicates 

```{r} 
sampleT_orig_f <- apply(assay_data_orig_f,2,sum)/10^6
sampleTDF_orig_f <- data.frame(sample=names(sampleT_orig_f), total=sampleT_orig_f)

sampleT_filt_genes_f <- apply(assay_data_filt_genes_f,2,sum)/10^6

sampleT_dup_f <- apply(assay_data_dup_f,2,sum)/10^6
sampleTDF_dup_f <- data.frame(sample=names(sampleT_dup_f), total=sampleT_dup_f)

duplicates.n <- 1 - (sampleTDF_dup_f$total/sampleTDF_orig_f$total)
duplicates_cat <- as.factor(ntile(duplicates.n, 3))
duplicates_cat <- ifelse(duplicates_cat == 1, "Low duplicate rate", ifelse(duplicates_cat == 2, "Medium duplicate rate", "High duplicate rate"))
duplicates_cat <- factor(duplicates_cat, levels = c("High duplicate rate", "Medium duplicate rate", "Low duplicate rate"))

```

#### 5.2.1.1 Original data

```{r}  
norm.expres.orig.d <- as.data.frame(apply(assay_data_orig_f, 1, function(x) (x/max(x, na.rm = TRUE))))
norm.expres.orig.d[is.na(norm.expres.orig.d)] <- 0
 
PCA.orig.d <- prcomp(norm.expres.orig.d,retx=TRUE, center=TRUE, scale=FALSE)
summary(PCA.orig.d)$importance[,c(1:3)] # PC1=0.24, PC2=0.09 and PC3=0.06

# Scores data
scores.data.orig.d <- data.frame(PCA.orig.d$x[, c("PC1", "PC2")])
scores.data.orig.d$duplicates <- as.factor(duplicates_cat)

scores.plot.orig.d <- ggplot(data = scores.data.orig.d, aes(x = PC1, y = PC2, colour = duplicates)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.orig.d)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.orig.d)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S2.3.1 PCA duplicates of the original data")+
  theme_bw()
ggplotly(scores.plot.orig.d)

```

#### 5.2.1.2 Data excluding lowly expressed genes that have high duplication ratios  

```{r}  
norm.expres.filt.genes.d <- as.data.frame(apply(assay_data_filt_genes_f, 1, function(x) (x/max(x, na.rm = TRUE))))
norm.expres.filt.genes.d[is.na(norm.expres.filt.genes.d)] <- 0
 
PCA.filt.genes.d <- prcomp(norm.expres.filt.genes.d,retx=TRUE, center=TRUE, scale=FALSE)
summary(PCA.filt.genes.d)$importance[,c(1:3)] # PC1=0.24, PC2=0.10 and PC3=0.06

# Scores data
scores.data.filt.genes.d <- data.frame(PCA.filt.genes.d$x[, c("PC1", "PC2")])
scores.data.filt.genes.d$duplicates <- as.factor(duplicates_cat)

scores.plot.filt.genes.d <- ggplot(data = scores.data.filt.genes.d, aes(x = PC1, y = PC2, colour = duplicates)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.filt.genes.d)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.filt.genes.d)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S2.3.2 PCA of the duplicates of the data excluding lowly expressed genes that have high duplication ratios")+
  theme(plot.title = element_text(size = 9))
ggplotly(scores.plot.filt.genes.d)

```

### 5.2.1.3 Data excluding duplicates

```{r}
norm.expres.dup.d <- as.data.frame(apply(assay_data_dup_f, 1, function(x) (x/max(x, na.rm = TRUE))))
norm.expres.dup.d[is.na(norm.expres.dup.d)] <- 0
 
PCA.dup.d <- prcomp(norm.expres.dup.d,retx=TRUE, center=TRUE, scale=FALSE)
summary(PCA.dup.d)$importance[,c(1:3)] # PC1=0.54, PC2=0.10 and PC3=0.05

# Scores data
scores.data.dup.d <- data.frame(PCA.dup.d$x[, c("PC1", "PC2")])
scores.data.dup.d$duplicates <- duplicates_cat

scores.plot.dup.d <- ggplot(data = scores.data.dup.d, aes(x = PC1, y = PC2, colour = duplicates)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.dup.d)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.dup.d)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S2.3.3 PCA duplicates of data excluding duplicates")+
  theme_bw()
ggplotly(scores.plot.dup.d)
```

### 5.2.2 Principal component Analysis of the duplicates and the total reads per sample 
#### 5.2.2.1 Original data 

```{r}
scores.data.orig.d$reads_per_sample <- sampleT_orig_f

scores.plot.orig.d <- ggplot(data = scores.data.orig.d, aes(x = PC1, y = PC2, colour = reads_per_sample, shape = duplicates)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.orig.d)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.orig.d)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S2.4.1 PCA of the duplicates and the total reads per sample of the original data")+
  theme(plot.title = element_text(size = 12))
ggplotly(scores.plot.orig.d)

shapiro.test(scores.data.orig.d$reads_per_sample) # The reads per sample in original data is normally distributed

anova_reads_original <- aov(reads_per_sample ~ duplicates, data = scores.data.orig.d)
summary(anova_reads_original)

posthoc_reads_original <- TukeyHSD(anova_reads_original)
print(posthoc_reads_original)

```

#### 5.2.2.2 Data excluding lowly expressed genes that have high duplication ratios  

```{r}  

scores.data.filt.genes.d$reads_per_sample <- sampleT_filt_genes

scores.plot.filt.genes.d <- ggplot(data = scores.data.filt.genes.d, aes(x = PC1, y = PC2, colour = reads_per_sample, shape = duplicates)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.filt.genes.d)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.filt.genes.d)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S2.4.2 PCA of the duplicates and the total reads per sample of the data excluding lowly expressed genes that have high duplication ratios")+
  theme(plot.title = element_text(size = 7))
ggplotly(scores.plot.filt.genes.d)

shapiro.test(scores.data.filt.genes.d$reads_per_sample) # The reads r sample in original data is normally distributed

anova_reads_filt_genes <- aov(reads_per_sample ~ duplicates, data = scores.data.filt.genes.d)
summary(anova_reads_filt_genes)

posthoc_reads_filt_genes_orig <- TukeyHSD(anova_reads_filt_genes)
print(posthoc_reads_filt_genes_orig)

```

#### 5.2.2.3 Data excluding duplicates

```{r}
scores.data.dup.d$reads_per_sample <- sampleT_dup_f

scores.plot.dup.d <- ggplot(data = scores.data.dup.d, aes(x = PC1, y = PC2, colour = reads_per_sample, shape = duplicates)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.dup.d)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.dup.d)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S2.4.3 PCA of the duplicates and the total reads per sample of data excluding duplicates")+
  theme(plot.title = element_text(size = 11))
ggplotly(scores.plot.dup.d)

shapiro.test(scores.data.dup.d$reads_per_sample) # The reads per sample in original data is not normally distributed

kruskal.test(scores.data.dup.d$reads_per_sample, scores.data.dup.d$duplicates)

# Subset the data for the two groups you want to compare
group_high <- scores.data.dup.d$reads_per_sample[scores.data.dup.d$duplicates == "High duplicate rate"]
group_medium <- scores.data.dup.d$reads_per_sample[scores.data.dup.d$duplicates == "Medium duplicate rate"]
group_low <- scores.data.dup.d$reads_per_sample[scores.data.dup.d$duplicates == "Low duplicate rate"]

# Perform the Mann-Whitney test
result_reads_high_low_dup <- wilcox.test(group_high, group_low)
print(result_reads_high_low_dup)

result_reads_high_medium_dup <- wilcox.test(group_high, group_medium)
print(result_reads_high_medium_dup)

result_reads_medium_low_dup <- wilcox.test(group_medium, group_low)
print(result_reads_medium_low_dup)

```


## 5.3 Data visualization of the difference in the total reads per sample after filtering

## 5.3.1 Original data  

```{r}
sampleT_orig_f
range(sampleT_orig_f) # 33.4-39.1
differences_orig <- sampleT_orig - sampleT_orig_f
range(differences_orig) # 0.0196-0.173

palette <- scale_fill_gradient(low = "#A9D6E5", high="#012A4A")
sampleTDF_orig_f<- data.frame(sample=names(sampleT_orig_f), total=differences_orig)

ggplot(sampleTDF_orig_f, aes(x = sample, y = total, fill = total)) +
  geom_bar(stat = "identity") +
  palette +
  labs(title = "Figure S2.5.1 Differences in the total reads per sample in the filtered original data", x = "Sample names", y = "Differences in the total reads per sample after filtering") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 6))
```

### 5.3.2 Data excluding lowly expressed genes that have high duplication ratios  

```{r}
sampleT_filt_genes_f
range(sampleT_filt_genes_f) # 33.3-39.1
differences_filt_genes <- sampleT_filt_genes - sampleT_filt_genes_f
range(differences_filt_genes) # 0.0178-0.169

sampleTDF_filt_genes_f<- data.frame(sample=names(sampleT_filt_genes_f), total=differences_filt_genes)

ggplot(sampleTDF_filt_genes_f, aes(x = sample, y = total, fill = total)) +
  geom_bar(stat = "identity") +
  palette +
  labs(title = "Figure S2.5.2 Differences in the reads per sample in the data excluding lowly expressed genes that have high duplication ratios", x = "Sample names", y = "Differences in the total reads per sample after filtering") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 6),
        plot.title = element_text(size = 15))
```
## 5.3.3 Data excluding duplicates 

```{r}
sampleT_dup_f
range(sampleT_dup_f) # 1.17-9.92
differences_dup <- sampleT_dup - sampleT_dup_f
range(differences_dup) # 0.00960-0.0689

sampleTDF_dup_f<- data.frame(sample=names(sampleT_dup), total=differences_dup)

ggplot(sampleTDF_dup_f, aes(x = sample, y = total, fill = total)) +
  geom_bar(stat = "identity") +
  palette +
  labs(title = "Figure S2.5.3 Differences in the total reads per sample in the filtered data excluding duplicates", x = "Sample names", y = "Differences in the total reads per sample after filtering") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 8))

```



# 6. Descriptive analysis of the duplicates
## 6.1 Available RNA

```{r}
available.RNA <- pData(ExpressionSet_orig)$available.ng
available.RNA <- as.factor(ifelse(available.RNA < 1000, "Limited RNA availability", "Non-limited RNA availability"))

fisher.availableRNA <- fisher.test(available.RNA, duplicates_cat)
print(fisher.availableRNA)

# Specify custom colors for each level
colors <- c("#013A63", "#2A6F97", "#61A5C2")

availableRNA.dup.df <- data.frame(available.RNA, duplicates_cat)

# Grouped bar plot with modified order and colors
ass.av.RNA <- ggplot(availableRNA.dup.df, aes(x = available.RNA, fill = duplicates_cat)) +
  geom_bar(position = "dodge") +
  xlab("Available RNA") +
  ylab("Frequency") +
  ggtitle("Figure S3.1 Association between available RNA and duplicate rate") +
  scale_fill_manual(values = colors)

print(ass.av.RNA)

```

## 6.1.1 Available RNA and timepoints

```{r}
timepoints <- pData(ExpressionSet_orig)$day

fisher.availableRNA.timepoints <- fisher.test(available.RNA, timepoints)
print(fisher.availableRNA.timepoints)

# Specify custom colors for each level
colors <- c("#013A63", "#61A5C2")

# Create a data frame with the timepoints and available RNA levels
 available.RNA.timepoints.df <- data.frame(timepoints, available.RNA)

# Create the grouped bar plot using ggplot2
ggplot(available.RNA.timepoints.df, aes(x = factor(timepoints), fill = available.RNA)) +
  geom_bar(position = "dodge") +
  scale_fill_manual(values = colors) +
  labs(x = "Timepoints", y = "Count", title = "Figure S3.1.1 Association between available RNA and timepoints",
       fill = "Available RNA") +
  theme_minimal()


```

## 6.2 RNA quality

```{r}
quality.rna <- pData(ExpressionSet_orig)$quality.rna

shapiro.test(quality.rna) # The reads per sample in original data is not normally distributed

kruskal.test(quality.rna, duplicates_cat)

RNAquality.dup.df <- data.frame(quality.rna, duplicates_cat)

# Boxplot with custom colors
ggplot(RNAquality.dup.df, aes(x = duplicates_cat, y = quality.rna, fill = duplicates_cat)) +
  geom_boxplot() +
  scale_fill_manual(values = c("#013A63", "#2A6F97", "#61A5C2")) +
  xlab("Duplicate Rate") +
  ylab("RNA quality") +
  ggtitle("Figure S3.2 Association between duplicate rate and RNA quality")

```

## 6.3 Contaminations 260/280

```{r}
nano260280 <- pData(ExpressionSet_orig)$nano260280

shapiro.test(nano260280) # The reads per sample in original data is not normally distributed

kruskal.test(nano260280, duplicates_cat)

# Subset the data for the two groups you want to compare
group_high_260280 <- nano260280[duplicates_cat == "High duplicate rate"]
group_medium_260280 <-  nano260280[duplicates_cat == "Medium duplicate rate"]
group_low_260280 <-  nano260280[duplicates_cat == "Low duplicate rate"]

# Perform the Mann-Whitney test
result_cont260180_high_low_dup <- wilcox.test(group_high_260280, group_low_260280)
print(result_cont260180_high_low_dup)

result_cont260280_high_med_dup <- wilcox.test(group_high_260280, group_medium_260280)
print(result_cont260280_high_med_dup)

result_cont260280_med_low_dup <- wilcox.test(group_medium_260280, group_low_260280)
print(result_cont260280_med_low_dup)

nano260280.dup.df <- data.frame(nano260280, duplicates_cat)

# Boxplot with custom colors
nano260280 <- ggplot(nano260280.dup.df, aes(x = duplicates_cat, y = nano260280, fill = duplicates_cat)) +
  geom_boxplot() +
  scale_fill_manual(values = c("#013A63", "#2A6F97", "#61A5C2")) +
  xlab("Duplicate Rate") +
  ylab("nano260280") +
  ggtitle("Figure S3.2 Association between duplicate rate and nanodrop 260/280")

print(nano260280)
```

## 6.4 Contaminations 260/230

```{r}
nano260230 <- pData(ExpressionSet_orig)$nano260230

shapiro.test(nano260230) # The reads per sample in original data is not normally distributed

kruskal.test(nano260230, duplicates_cat)

# Subset the data for the two groups you want to compare
group_high_260230 <- nano260230[duplicates_cat == "High duplicate rate"]
group_medium_260230 <-  nano260230[duplicates_cat == "Medium duplicate rate"]
group_low_260230 <-  nano260230[duplicates_cat == "Low duplicate rate"]

# Perform the Mann-Whitney test
result_cont260230_high_low_dup <- wilcox.test(group_high_260230, group_low_260230)
print(result_cont260230_high_low_dup)

result_cont260230_high_medium_dup <- wilcox.test(group_high_260230, group_medium_260230)
print(result_cont260230_high_medium_dup)

result_cont260230_medium_low_dup <- wilcox.test(group_medium_260230, group_low_260230)
print(result_cont260230_medium_low_dup)

nano260230.dup.df <- data.frame(nano260230, duplicates_cat)

# Boxplot with custom colors
nano260230 <- ggplot(nano260230.dup.df, aes(x = duplicates_cat, y = nano260230, fill = duplicates_cat)) +
  geom_boxplot() +
  scale_fill_manual(values = c("#013A63", "#2A6F97", "#61A5C2")) +
  xlab("Duplicate Rate") +
  ylab("nano260230") +
  ggtitle("Figure S3.3 Association between duplicate rate and nanodrop 260/230")

print(nano260230)

```
## 6.5 Library preparation batch

```{r}
library.batch <- pData(ExpressionSet_orig)$library.batch

fisher.library.batch <- fisher.test(library.batch, duplicates_cat)
print(fisher.library.batch)

```

## 6.6 Treatment

```{r}
treatment <- as.factor(pData(ExpressionSet_orig)$treat)

fisher.treatment<- fisher.test(treatment, duplicates_cat)
print(fisher.treatment)

```

## 6.7 Timepoints

```{r}
timepoints <- as.factor(pData(ExpressionSet_orig)$day)

fisher.timepoints<- chisq.test(timepoints, duplicates_cat)
print(fisher.timepoints)

```

## 6.8 RNA extraction batch

```{r}
extraction.batch <- as.factor(pData(ExpressionSet_orig)$extraction.batch)

fisher.extr.batch<- fisher.test(extraction.batch, duplicates_cat)
print(fisher.extr.batch)

```

## 6.9 Sex

```{r}
sex <- as.factor(pData(ExpressionSet_orig)$sex)

fisher.sex <- fisher.test(sex, duplicates_cat)
print(fisher.sex)

```

## 6.10 Ages

```{r}
ages <- as.numeric(pData(ExpressionSet_orig)$age)

age_groups <- ifelse(ages<=26, "Young", "Adult")

fisher.age <- fisher.test(age_groups, duplicates_cat)
print(fisher.age)

```

## 6.11 Patients

```{r}
patients <- as.factor(pData(ExpressionSet_orig)$studyno)
fisher.patients <- fisher.test(patients, duplicates_cat)
print(fisher.patients)

# Create a contingency table of patients and duplicates
contingency_table <- table(patients, duplicates_cat)

# Create a mosaic plot
mosaicplot(contingency_table, main = "Figure S3.4 Association between patients and duplicates",
           xlab = "Patients", ylab = "Duplicates",
           color = c("#013A63", "#2A6F97", "#61A5C2"))
```
# 7. TMM Normalization
## 7.1 Original data

```{r}
DGEListOrig <- DGEList(counts = assay_data_orig_f)
Norm.FactorOrig <- calcNormFactors(DGEListOrig, method = "TMM")
countsTMMOrig <- cpm(Norm.FactorOrig, log = T)

palette_TMM <- c("#A9D6E5", "#9ECDDF", "#94C8DC", "#89C2D9", "#66A2BD", "#5799b6", "#2C7DA0", "#2B769C", "#2A6F97", "#206793", "#165F8F", "#0C578B", "#075389",  "#01497C",  "#014270", "#013A63", "#013257", "#012A4A")
              
hist(countsTMMOrig[,1], xlab="log2-ratio", col = palette_TMM, main="Figure S4.1 TMM normalization of original data") 

```

## 7.2 Data excluding lowly expressed genes that have high duplication ratios 

```{r}
DGEListfiltgenes <- DGEList(counts = assay_data_filt_genes_f)
Norm.FactorFiltGenes <- calcNormFactors(DGEListfiltgenes, method = "TMM")
countsTMMFiltGenes <- cpm(Norm.FactorFiltGenes, log = T)
              
hist(countsTMMFiltGenes[,1], xlab="log2-ratio", col = palette_TMM, main="Figure S4.2 TMM normalization of the data excluding lowly expressed genes that have high duplication ratios") 

```

### 7.3. Data excluding duplicates

```{r}
DGEListDup <- DGEList(counts = assay_data_dup_f)
Norm.FactorDup <- calcNormFactors(DGEListDup, method = "TMM")
countsTMMDup <- cpm(Norm.FactorDup, log = T)

hist(countsTMMDup[,1], xlab="log2-ratio", col = palette_TMM, main="Figure S4.3 TMM normalization of data excluding duplicates") 
```


# 8. Unsupervised Analysis: Principal Component Analysis
## 8.1. Duplications
### 8.1.1. Original data

```{r}  
norm.expres.orig <- as.data.frame(apply(countsTMMOrig, 1, function(x) (x/max(x, na.rm = TRUE))))
norm.expres.orig[is.na(norm.expres.orig)] <- 0
 
PCA.orig <- prcomp(norm.expres.orig,retx=TRUE, center=TRUE, scale=FALSE)
summary(PCA.orig)$importance[,c(1:3)] # PC1=0.26, PC2=0.14 and PC3=0.09

# Scores data
scores.data.orig <- data.frame(PCA.orig$x[, c("PC1", "PC2")])
scores.data.orig$duplicates <- as.factor(duplicates_cat)

scores.plot.orig <- ggplot(data = scores.data.orig, aes(x = PC1, y = PC2, colour = duplicates)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.orig)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.orig)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.1.1 PCA of the duplicates from the normalized original data")+
  theme_bw()
ggplotly(scores.plot.orig)
```

### 8.1.2 Data excluding lowly expressed genes that have high duplication ratios 

```{r}  
norm.expres.filt.genes <- as.data.frame(apply(countsTMMFiltGenes, 1, function(x) (x/max(x, na.rm = TRUE))))
norm.expres.filt.genes[is.na(norm.expres.filt.genes)] <- 0
 
PCA.filt.genes <- prcomp(norm.expres.filt.genes,retx=TRUE, center=TRUE, scale=FALSE)
summary(PCA.filt.genes)$importance[,c(1:3)] # PC1=0.49, PC2=0.24 and PC3=0.09

# Scores data
scores.data.filt.genes <- data.frame(PCA.filt.genes$x[, c("PC1", "PC2")])
scores.data.filt.genes$duplicates <- as.factor(duplicates_cat)

scores.plot.filt.genes <- ggplot(data = scores.data.filt.genes, aes(x = PC1, y = PC2, colour = duplicates)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.filt.genes)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.filt.genes)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.1.2 PCA of the duplicates from the normalized data excluding lowly expressed genes that have high duplication ratios")+
  theme(plot.title = element_text(size = 8))
ggplotly(scores.plot.filt.genes)
```

### 8.1.3 Data excluding duplicates

```{r}
norm.expres.dup <- as.data.frame(apply(countsTMMDup, 1, function(x) (x/max(x, na.rm = TRUE))))
norm.expres.dup[is.na(norm.expres.dup)] <- 0
 
PCA.dup <- prcomp(norm.expres.dup,retx=TRUE, center=TRUE, scale=FALSE)
summary(PCA.dup)$importance[,c(1:3)] # PC1=0.15, PC2=0.24 and PC3=0.06

# Scores data
scores.data.dup <- data.frame(PCA.dup$x[, c("PC1", "PC2")])
scores.data.dup$duplicates <- duplicates_cat

scores.plot.dup <- ggplot(data = scores.data.dup, aes(x = PC1, y = PC2, colour = duplicates)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.dup)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.dup)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.1.3 PCA of the duplicates from the normalized data excluding the duplicates")+
  theme(plot.title = element_text(size = 12))
ggplotly(scores.plot.dup)
```

## 8.2 Treatment
### 8.2.1 Original data

```{r}  
scores.data.orig$treatment <-ifelse(treatment==1, "Treatment", "Control")
  
scores.plot.orig <- ggplot(data = scores.data.orig, aes(x = PC1, y = PC2, colour = treatment)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.orig)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.orig)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.2.1 PCA of the treatment from the normalized original data")+
  theme_bw()
ggplotly(scores.plot.orig)
```

### 8.2.2 Data excluding lowly expressed genes that have high duplication ratios 

```{r}  
scores.data.filt.genes$treatment <-ifelse(treatment==1, "Treatment", "Control")
  
scores.plot.filt.genes <- ggplot(data = scores.data.filt.genes, aes(x = PC1, y = PC2, colour = treatment)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.filt.genes)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.filt.genes)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.2.2 PCA of the treatment from the normalized data excluding lowly expressed genes that have high duplication ratios")+
  theme(plot.title = element_text(size = 7))
ggplotly(scores.plot.filt.genes)
```

### 8.2.3 Data excluding duplicates

```{r}
scores.data.dup$treatment <-ifelse(treatment==1, "Treatment", "Control")

scores.plot.dup <- ggplot(data = scores.data.dup, aes(x = PC1, y = PC2, colour = treatment)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.dup)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.dup)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.2.3 PCA of the treatment from the normalized data excluding the duplicates")+
  theme(plot.title = element_text(size = 12))
ggplotly(scores.plot.dup)
```


## 8.3 Treatment and timepoints
### 8.3.1 Original data

```{r}  
  
scores.plot.orig <- ggplot(data = scores.data.orig, aes(x = PC1, y = PC2, colour = treatment, shape = timepoints)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.orig)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.orig)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.3.1 PCA of the treatment and the timepoints from the normalized original data")+
  theme(plot.title = element_text(size = 12))
ggplotly(scores.plot.orig)
```

### 8.3.2 Data excluding lowly expressed genes that have high duplication ratios 

```{r}  
scores.plot.filt.genes <- ggplot(data = scores.data.filt.genes, aes(x = PC1, y = PC2, colour = treatment, shape = timepoints)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.filt.genes)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.filt.genes)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.3.2 PCA of the treatment and timepoints from the normalized data excluding lowly expressed genes that have high duplication ratios")+
  theme(plot.title = element_text(size = 7))
ggplotly(scores.plot.filt.genes)
```

### 8.3.3 Data excluding duplicates

```{r}

scores.plot.dup <- ggplot(data = scores.data.dup, aes(x = PC1, y = PC2, colour = treatment, shape = timepoints)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.dup)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.dup)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.3.3 PCA of the treatment and the timepoints from the normalized data excluding the duplicates")+
  theme(plot.title = element_text(size = 10))
ggplotly(scores.plot.dup)
```

## 8.4 Timepoints

### 8.4.1 Original data

```{r}  
scores.plot.orig <- ggplot(data = scores.data.orig, aes(x = PC1, y = PC2, colour = timepoints)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.orig)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.orig)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.4.1 PCA of the timepoints from the normalized original data")+
  theme_bw()
ggplotly(scores.plot.orig)
```

### 8.4.2 Data excluding lowly expressed genes that have high duplication ratios 

```{r}  
scores.plot.filt.genes <- ggplot(data = scores.data.filt.genes, aes(x = PC1, y = PC2, colour = timepoints)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.filt.genes)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.filt.genes)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.4.2 PCA of the timepoints from the normalized data excluding lowly expressed genes that have high duplication ratios")+
  theme(plot.title = element_text(size = 7))
ggplotly(scores.plot.filt.genes)
```

### 8.4.3 Data excluding lowly expressed genes that have high duplication ratios 

```{r}

scores.plot.dup <- ggplot(data = scores.data.dup, aes(x = PC1, y = PC2, colour = timepoints)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.dup)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.dup)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.4.3 PCA of the timepoints from the normalized data excluding the duplicates")+
  theme(plot.title = element_text(size = 12))
ggplotly(scores.plot.dup)
```


## 8.5 Sex and timepoints
### 8.5.1 Original data

```{r}  
scores.data.orig$sex <-ifelse(sex==1, "Female", "Male")
  
scores.plot.orig <- ggplot(data = scores.data.orig, aes(x = PC1, y = PC2, colour = sex, shape = timepoints)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.orig)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.orig)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.5.1 PCA of the sex and the timepoints from the normalized original data")+
  theme_bw()
ggplotly(scores.plot.orig)
```

### 8.5.2 Data excluding lowly expressed genes that have high duplication ratios 

```{r}  
scores.data.filt.genes$sex <- ifelse(sex==1, "Female", "Male")

scores.plot.filt.genes <- ggplot(data = scores.data.filt.genes, aes(x = PC1, y = PC2, colour = sex, shape=timepoints)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.filt.genes)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.filt.genes)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.5.2 PCA of the sex and the timepoints from the normalized data excluding lowly expressed genes that have high duplication ratios")+
  theme(plot.title = element_text(size = 6))
ggplotly(scores.plot.filt.genes)
```

### 8.5.3 Data excluding duplicates

```{r}
scores.data.dup$sex <-ifelse(sex==1, "Female", "Male")

scores.plot.dup <- ggplot(data = scores.data.dup, aes(x = PC1, y = PC2, colour = sex, shape = timepoints)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.dup)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.dup)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.5.3 PCA of the sex and the timepoints of the normalized data excluding the duplicates")+
  theme(plot.title = element_text(size = 11))
ggplotly(scores.plot.dup)
```

## 8.6 Patients and timepoints
### 8.6.1 Original data

```{r}  

scores.plot.orig <- ggplot(data = scores.data.orig, aes(x = PC1, y = PC2, colour = patients, shape = timepoints)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.orig)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.orig)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.6.1 PCA of the patients and the timepoints from the normalized original data")+
  theme(plot.title = element_text(size = 11))
ggplotly(scores.plot.orig)
```


### 8.6.2 Data excluding lowly expressed genes that have high duplication ratios 

```{r}  
scores.plot.filt.genes <- ggplot(data = scores.data.filt.genes, aes(x = PC1, y = PC2, colour = patients, shape=timepoints)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.filt.genes)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.filt.genes)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.6.2 PCA of the patients and the timepoints from the normalized data excluding lowly expressed genes that have high duplication ratios")+
  theme(plot.title = element_text(size = 7))
ggplotly(scores.plot.filt.genes)
```

### 8.6.3 Data excluding duplicates

```{r}

scores.plot.dup <- ggplot(data = scores.data.dup, aes(x = PC1, y = PC2, colour = patients, shape = timepoints)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.dup)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.dup)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.6.3 PCA of the patients and the timepoints from the normalized data excluding the duplicates")+
  theme(plot.title = element_text(size = 9))
ggplotly(scores.plot.dup)
```

## 8.7 Extraction batch
### 8.7.1 Original data

```{r}  
extraction.batch <- as.factor(pData(ExpressionSet_orig)$extraction.batch)
scores.plot.orig <- ggplot(data = scores.data.orig, aes(x = PC1, y = PC2, colour = extraction.batch)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.orig)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.orig)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.7.1 PCA of the extraction batch from the normalized original data")+
  theme_bw()
ggplotly(scores.plot.orig)
```

### 8.7.2 Data excluding lowly expressed genes that have high duplication ratios 

```{r}  

scores.plot.filt.genes <- ggplot(data = scores.data.filt.genes, aes(x = PC1, y = PC2, colour = extraction.batch)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.filt.genes)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.filt.genes)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.7.2 PCA of the extraction batch from the normalized data excluding lowly expressed genes that have high duplication ratios")+
  theme(plot.title = element_text(size = 8))
ggplotly(scores.plot.filt.genes)
```

### 8.7.3 Data excluding duplicates

```{r}

scores.plot.dup <- ggplot(data = scores.data.dup, aes(x = PC1, y = PC2, colour = extraction.batch)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.dup)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.dup)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.7.3 PCA of the extraction batch of the normalized data excluding the duplicates")+
  theme(plot.title = element_text(size = 12))

ggplotly(scores.plot.dup)
```

## 8.8 Ages
### 8.8.1 Original data

```{r}  
ages <- as.factor(pData(ExpressionSet_orig)$age)
scores.data.orig$ages <- as.factor(ntile(ages, 3))

scores.plot.orig <- ggplot(data = scores.data.orig, aes(x = PC1, y = PC2, colour = ages)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.orig)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.orig)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.8.1 PCA of the ages from the normalized original data")+
  theme_bw()
ggplotly(scores.plot.orig)
```

### 8.8.2 Data excluding lowly expressed genes that have high duplication ratios 

```{r}  
scores.data.filt.genes$ages <- as.factor(ntile(ages, 3))

scores.plot.filt.genes <- ggplot(data = scores.data.filt.genes, aes(x = PC1, y = PC2, colour = ages)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.filt.genes)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.filt.genes)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.8.2 PCA of the ages from the normalized data excluding lowly expressed genes that have high duplication ratios")+
  theme(plot.title = element_text(size = 7))
ggplotly(scores.plot.filt.genes)
```

### 8.8.3 Data excluding duplicates

```{r}
scores.data.dup$ages <-  as.factor(ntile(ages, 3))
scores.plot.dup <- ggplot(data = scores.data.dup, aes(x = PC1, y = PC2, colour = ages)) +
  geom_point(alpha = I(0.7), size = 4) +
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  xlab(paste("PC1 (", round(summary(PCA.dup)$importance[2,1], 2) * 100, "%)"))+
  ylab(paste("PC2 (", round(summary(PCA.dup)$importance[2,2], 2) * 100, "%)"))+
  stat_ellipse() +
  labs(title = "Figure S5.8.3 PCA of the ages of the normalized data excluding the duplicates")+
  theme_bw()
ggplotly(scores.plot.dup)
```

